### PR TITLE
Change CI tooling

### PR DESCRIPTION
**Pull request by LeePorte**
What
----

As we are coalescing around GitHub Actions and there is likely to be a fair number of tech docs template based repos that are going to be doing similar it makes sense to do this with our own first.

How to review
-------------

Check the tests pass and a code review

Who can review
--------------

Anyone

### CODE REVIEW

File: .github/workflows/pr-tests.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+name: pr-tests
+on: pull_request
+
+env:
+  NODE_VERSION: "14.x"
+  RUBY_VERSION: "2.7.1"
```
Comment (kr8n3r) [inline .github/workflows/pr-tests.yml:7]: 2.7.2

File: .github/workflows/pr-tests.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: ruby/setup-ruby@08245253a76fa4d1e459b7809579c62bd9eb718a #v1.111.0
+        with:
+          ruby-version: "${{env.RUBY_VERSION}}"
+          bundler-cache: true
+
+      - uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08 #v4.1.0
+        with:
+          python-version: "${{env.PYTHON_VERSION}}"
+
+      - run: npm ci
```
Comment (kr8n3r) [inline .github/workflows/pr-tests.yml:32]: we don't have any npm dependecies on this one

File: .github/workflows/pr-tests.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/setup-node@2fddd8803e2f5c9604345a0b591c3020ee971a93 #v3.4.1
+        with:
+          node-version: "${{env.NODE_VERSION}}"
+
+      - uses: ruby/setup-ruby@08245253a76fa4d1e459b7809579c62bd9eb718a #v1.111.0
+        with:
+          ruby-version: "${{env.RUBY_VERSION}}"
+          bundler-cache: true
+
+      - uses: actions/setup-python@c4e89fac7e8767b327bbad6cb4d859eda999cf08 #v4.1.0
```
Comment (kr8n3r) [inline .github/workflows/pr-tests.yml:28]: i think python is only if you want to run locally, don't think it ran on circle CI

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+concurrency: production
```
[the remark this answers is not in the document]
Comment (LeePorte) [reply .github/workflows/deploy.yml:9]: `When a concurrent job or workflow is queued, if another job or workflow using the same concurrency group in the repository is in progress, the queued job or workflow will be pending.`

from https://docs.github.com/en/actions/using-jobs/using-concurrency

**Review by kr8n3r — APPROVED** [review]
looks good now. we 'll see the deployment in action once this is merged

**Pull request merged**